pkg/virt-api/rest/subresource.go

```diff
@@ -419,6 +422,19 @@ func (app *SubresourceAPIApp) RestartVMRequestHandler(request *restful.Request,
 	name := request.PathParameter("name")
 	namespace := request.PathParameter("namespace")
 
+	bodyStruct := &v1.RestartOptions{}
+
+	if request.Request.Body != nil {
+		err := yaml.NewYAMLOrJSONDecoder(request.Request.Body, 1024).Decode(&bodyStruct)
+		switch err {
+		case io.EOF, nil:
+			break
+		default:
+			response.WriteError(http.StatusForbidden, fmt.Errorf("Can not unmarshal Request body to struct, error: %s", err))
+			return
+		}
+	}
+
```

Comment:
I know  you did not write the code below, but where it returns a `StatusForbidden`, it should actually return a `StatusConflict`.